src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -379,10 +379,9 @@ private void NotifyTransactionFailureAndRerunIndependently(List<MergedTransactio
                                 {
                                     throw;
                                 }
-                                NotifyTransactionFailureAndRerunIndependently(currentPendingOps, e);
-                                return;
                             }
-                          
+                            NotifyTransactionFailureAndRerunIndependently(currentPendingOps, e);
```

Comment:
can you explain this change?

Reply:
It is not necessary related to RavenDB-9917, since I wasn't able to reproduce it.

But this change prevent opening a new transaction in an already opened one.
If we had an async transaction that failed, we are going to try and run it one by one.
For this matter we will open a transaction for each pendingOp. So we need first to dispose the failed transaction before opening a new one.